consensus/cryptarchia-engine/src/time.rs

```diff
@@ -16,6 +21,20 @@ impl Slot {
     pub fn genesis() -> Self {
         Self(0)
     }
+
+    pub fn from_offset_and_config(
+        offset_date_time: OffsetDateTime,
+        slot_config: SlotConfig,
+    ) -> Self {
+        // TODO: leap seconds / weird time stuff
+        let since_start = offset_date_time - slot_config.chain_start_time;
+        if since_start.is_negative() {
+            // current slot is behind the start time, so return default 0
+            Slot::genesis()
+        } else {
+            Slot::from(since_start.whole_seconds() as u64 / slot_config.slot_duration.as_secs())
```

Comment:
This could still underflow, right? Are we leaving them like this, and address them in a different PR where we introduce Clippy lints?

Reply:
No, I would prefer to address them here with context. Problem is the same, values should come well from config.

Reply:
For reference, changes will be integrated later [here](https://github.com/logos-co/nomos-node/pull/1090)